tests/components/modbus/test_light.py

```diff
@@ -314,9 +338,101 @@ async def test_light_service_turn(
                 {
                     CONF_NAME: TEST_ENTITY_NAME,
                     CONF_ADDRESS: 1234,
+                    CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                    CONF_SCAN_INTERVAL: 0,
+                    CONF_BRIGHTNESS_REGISTER: 1,
+                    CONF_COLOR_TEMP_REGISTER: 2,
+                },
+                {
+                    CONF_NAME: f"{TEST_ENTITY_NAME} 2",
+                    CONF_ADDRESS: 1234,
+                    CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                    CONF_SCAN_INTERVAL: 0,
+                    CONF_BRIGHTNESS_REGISTER: 1,
+                },
+            ]
+        }
+    ],
+)
+async def test_color_temp_light(
+    hass: HomeAssistant, mock_modbus_ha, modbus_light_entity
+) -> None:
+    """Test Modbus Light color temperature amd brightness."""
+    assert hass.states.get(ENTITY_ID).state == STATE_OFF
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        service_data={
+            ATTR_ENTITY_ID: ENTITY_ID,
+            ATTR_BRIGHTNESS: 128,
+            ATTR_COLOR_TEMP_KELVIN: 2000,
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get(ENTITY_ID).state == STATE_ON
+    calls = mock_modbus_ha.write_register.call_args_list
+    expected_modbus_brightness = 50
+    expected_modbus_temp = 0
```

Comment:
lets also move this to the test params, since it should not be the same for both lights